File: Programmers/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        Arrays.sort(lottos);
        Arrays.sort(win_nums);
        
        int zero = 0;
        int winning = 0;
        
        for (int lotto : lottos) {
            if (lotto == 0) {
                zero++;
                continue;
            }
            
            if (IntStream.of(win_nums).anyMatch(n -> n == lotto)) {
                winning++;
            }
        }
        
        return new int[] {Math.min(6, 6 - (winning + zero) + 1), Math.min(6, 6 - winning + 1)};
    }
}

File: Programmers/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    sort(lottos.begin(), lottos.end());
    sort(win_nums.begin(), win_nums.end());
    
    int zero = count(lottos.begin(), lottos.end(), 0);
    int winning = 0;
    
    for (int lotto : lottos)
    {
        if (lotto == 0) 
            continue;
        
        if (find(win_nums.begin(), win_nums.end(), lotto) != win_nums.end())
        {
            winning++;
        }
    }
    
    int maxRank = min(6, 6 - (winning + zero) + 1);
    int minRank = min(6, 6 - winning + 1);
    
    return {maxRank, minRank};
}